controllers/managedocs_controller.go

```diff
@@ -1338,6 +1338,25 @@ func (r *ManagedOCSReconciler) reconcileMonitoringResources() error {
 	for i := range promRuleList.Items {
 		obj := promRuleList.Items[i]
 		utils.AddLabel(obj, monLabelKey, monLabelValue)
+		// add an annotation "managed: true" to the prometheus rule with the label "prometheus: rook-prometheus"
+		if v, ok := obj.GetLabels()["prometheus"]; ok && v == "rook-prometheus" {
+			// remove unwanted fields from the prometheus rule
+			unwantedRuleAnnotations := []string{"documentation"}
+			groups := obj.Spec.Groups
+			for groupIndex := range groups {
+				group := &groups[groupIndex]
+				for ruleIndex := range group.Rules {
+					rule := &group.Rules[ruleIndex]
+					if rule.Alert != "" {
+						for _, annotation := range unwantedRuleAnnotations {
+							delete(rule.Annotations, annotation)
+						}
+					}
+				}
+			}
+			// disallow prometheus rule reconciliation by ceph cluster
+			utils.AddAnnotation(obj, "managed", "true")
```

Comment:
- we already have a standard key & value for monitoring resources, I believe it's best to re-use that for annotations as well
- https://github.com/red-hat-storage/ocs-osd-deployer/blob/6bba26e7609257de451f66878a3c9ce1fe674170/controllers/managedocs_controller.go#L95-L96

Reply:
Sure, I'll remove this (since its already added in `reconcileMonitoringResources`) and make the commit message more details.

Reply:
Done.